### PR TITLE
feat(console): website resource

### PR DESCRIPTION
**Pull request by ainvoner**
add website resource support to the console.
users can open the website url in a different browser tab from the website resource interaction view

resolves: https://github.com/winglang/wing/issues/2973

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by ekeren** [comment]
@ainvoner , can you add an image?

File: apps/wing-console/console/server/src/router/website.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {createProcedure, createRouter} from "../utils/createRouter";
```
Comment (polamoros) [inline apps/wing-console/console/server/src/router/website.ts:1]: missing prettier on this file

File: apps/wing-console/console/ui/src/features/resource-interaction-view.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { ScheduleInteractionView } from "./schedule-interaction-view.js";
 import { TableInteractionView } from "./table-interaction-view.js";
 import { TopicInteractionView } from "./topic-interaction-view.js";
 import { UnsupportedInteractionView } from "./unsupported-interaction-view.js";
+import {WebsiteInteractionView} from "./website-interaction-view";
```
Comment (polamoros) [inline apps/wing-console/console/ui/src/features/resource-interaction-view.tsx:11]:
```suggestion
import { WebsiteInteractionView } from "./website-interaction-view";
```

File: apps/wing-console/console/ui/src/features/website-interaction-view.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+}
+export const WebsiteInteractionView = ({resourcePath}: WebsiteInteractionViewProps) => {
+    const { appMode } = useContext(AppContext);
+    const {open} = useOpenExternal();
+    const {url} = useWebsite({resourcePath});
+
+    return (
+        <WebsiteInteraction appMode={appMode} onUrlClick={open} url={url} />
+    );
+}
```
Comment (polamoros) [inline apps/wing-console/console/ui/src/features/website-interaction-view.tsx:18]:
```suggestion
import { Attribute, useTheme } from "@wingconsole/design-system";
import { ArrowTopRightOnSquareIcon } from "@heroicons/react/24/solid";
import { AppMode } from "../AppContext.js";
import classNames from "classnames";

export interface WebsiteInteractionProps {
  url: string;
  appMode: AppMode;
  onUrlClick: (url: string) => void;
}
export const WebsiteInteraction = ({
  appMode,
  url,
  onUrlClick,
}: WebsiteInteractionProps) => {
  const { theme } = useTheme();
  return (
    <div className="h-full flex-1 flex flex-col text-sm space-y-1">
      <div className="relative grow flex-row flex items-center">
        {appMode === "local" && (
          <>
            <Attribute name="URL" value={url} noLeftPadding></Attribute>
            <ArrowTopRightOnSquareIcon
              className={classNames(
                theme.text2,
                "text-sm flex ml-2 h-4 w-4 cursor-pointer",
              )}
              onClick={() => onUrlClick(url)}
            />
          </>
        )}
      </div>
    </div>
  );
};

```

File: apps/wing-console/console/ui/src/services/use-website.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    const [url, setUrl] = useState("");
+
+    useEffect(() => {
+        setUrl(websiteUrl.data ?? "");
+    }, [websiteUrl.data]);
+
+    return {
+        url
+    }
+}
```
Comment (polamoros) [inline apps/wing-console/console/ui/src/services/use-website.ts:18]:
```suggestion
import { trpc } from "./trpc.js";
import { useEffect, useState } from "react";

export interface UseWebsiteOptions {
  resourcePath: string;
}
export const useWebsite = ({ resourcePath }: UseWebsiteOptions) => {
  const websiteUrl = trpc["website.url"].useQuery({ resourcePath });
  const [url, setUrl] = useState("");

  useEffect(() => {
    setUrl(websiteUrl.data ?? "");
  }, [websiteUrl.data]);

  return {
    url,
  };
};
```

**Comment by polamoros** [comment]
What about moving the button inside the url container?

image
vs
image

**Review by polamoros — APPROVED** [review]

**Pull request merged**